frontend/src/metabase/sharing/components/SlackChannelField.jsx

```diff
@@ -0,0 +1,76 @@
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+import { t } from "ttag";
+
+import AutocompleteInput from "metabase/core/components/AutocompleteInput";
+
+const getPrivateChannelWarning = () =>
+  t`In order to send subscriptions and alerts to private Slack channels, you must first add the Metabase bot to them.`;
+
+const CHANNEL_FIELD_NAME = "channel";
+const CHANNEL_PREFIX = "#";
+const USER_PREFIX = "@";
+
+const ALLOWED_PREFIXES = [CHANNEL_PREFIX, USER_PREFIX];
+
+function SlackChannelField({ channel, channelSpec, onChannelPropertyChange }) {
+  const [warning, setWarning] = useState(null);
```

Comment:
nit: it seems it can be changed to a boolean field, e.g. if the warning is visible